python/test_dcr_template.py

```diff
@@ -0,0 +1,629 @@
+        test_reg = np.load(data_file)
+        self.assertFloatsEqual(reg, test_reg)
+
+    def test_frequency_derivative_regularization(self):
+        data_file = "test_data/frequency_derivative_regularization.npy"
+        reg = DcrCorrection._build_regularization(x_size=self.kernel_size, y_size=self.kernel_size,
+                                                  n_step=self.n_step, frequency_second_regularization=True)
+        test_reg = np.load(data_file)
+        self.assertFloatsEqual(reg, test_reg)
+
+    def test_multiple_regularization(self):
+        spatial_file = "test_data/spatial_regularization.npy"
+        freq_file = "test_data/frequency_regularization.npy"
+        deriv_file = "test_data/frequency_derivative_regularization.npy"
+
+        reg = DcrCorrection._build_regularization(x_size=self.kernel_size, y_size=self.kernel_size,
+                                                  n_step=self.n_step, spatial_regularization=True,
+                                                  frequency_regularization=True,
+                                                  frequency_second_regularization=True)
+        freq_reg = np.append(np.load(freq_file), np.load(deriv_file), axis=1)
+        test_reg = np.append(np.load(spatial_file), freq_reg, axis=1)
+        self.assertFloatsEqual(reg, test_reg)
+
+
+class SolverTestCast(lsst.utils.tests.TestCase):
+    def setUp(self):
+        data_file = "test_data/exposures.npy"
+        exposures = np.load(data_file)
+        self.dcrCorr = _BasicDcrCorrection(band_name='g', n_step=3, kernel_size=5, exposures=exposures)
+
+    def tearDown(self):
+        del self.dcrCorr
+
+    def test_build_dcr_kernel_full(self):
+        data_file = "test_data/build_dcr_kernel_full_vals.npy"
+        kernel = self.dcrCorr._build_dcr_kernel(use_full=True)
+        kernel_ref = np.load(data_file)
+        self.assertFloatsEqual(kernel, kernel_ref)
+
+    def test_build_dcr_kernel(self):
+        data_file = "test_data/build_dcr_kernel_vals.npy"
+        kernel = self.dcrCorr._build_dcr_kernel(use_full=False)
+        kernel_ref = np.load(data_file)
+        self.assertFloatsEqual(kernel, kernel_ref)
+
+    def test_build_model(self):
+        """Call build_model with as many options as possible turned off."""
+        data_file = "test_data/build_model_vals.npy"
+        self.dcrCorr.psf_avg = 1.
+        self.dcrCorr.build_model(use_full=False, use_regularization=False,
+                                 use_only_detected=False, verbose=False)
+        model_vals = self.dcrCorr.model
+        model_ref = np.load(data_file)
+        self.assertFloatsEqual(model_vals, model_ref)
+
+    def test_solve_model_no_regularization(self):
+        data_file = "test_data/solve_model_vals.npy"
+        y_size, x_size = self.dcrCorr.exposures[0].getDimensions()
+        pix_radius = self.dcrCorr.kernel_size//2
+        # Make j and i different slightly so we can tell if the indices get swapped
```

Comment:
Good idea, but it's not obvious you are doing this. i and j are computed from the exposure dimensions, and I have no idea what those are.